core/domain/user_jobs_one_off.py

```diff
@@ -468,6 +468,184 @@ def reduce(key, values):
         yield (key, len(values))
 
 
+class FixUserSettingsCreatedOnOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that fixes the invalid values of created_on attribute in the
+    UserSettingsModel.
+
+    This is a one-off job and can be removed from the codebase
+    after we resolve this issue by running the job once.
+    """
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        super(FixUserSettingsCreatedOnOneOffJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(user_settings_model):
+        user_id = user_settings_model.id
+        user_dates_list = [
+            ('UserSettingsModel', user_settings_model.created_on),
+            ('UserSettingsModel', user_settings_model.last_updated),
+            ('UserSettingsModel', user_settings_model.last_agreed_to_terms),
+            ('UserSettingsModel', user_settings_model.last_logged_in),
+            ('UserSettingsModel', user_settings_model.first_contribution_msec),
+            (
+                'UserSettingsModel',
+                user_settings_model.last_started_state_editor_tutorial
+            ),
+            (
+                'UserSettingsModel',
+                user_settings_model.last_started_state_translation_tutorial
+            ),
+            (
+                'UserSettingsModel',
+                user_settings_model.last_edited_an_exploration
+            ),
+            (
+                'UserSettingsModel',
+                user_settings_model.last_created_an_exploration
+            ),
+        ]
+
+        # Models in user storage module keyed by user_id.
+        completed_activities_model = (
```

Comment:
Please use fetch_multiple_entities_by_ids_and_models() in core/platform/datastore for this instead, rather than getting each model one at a time. The latter is very time/resource-consuming, especially given that you're running this repeatedly (once for each user).

Reply:
Did the change now. Thanks!